### ISSUE TITLE
installing librealsense on Ubuntu 22.04 using vcpkg 

### ISSUE THREAD

**Issue by monajalal** [microsoft/vcpkg#28449]
Hi,

I have a D435 camera and want to install librealsense and got the following error. Could you please guide me how to fix this?

```
[mona@ARD-WS-24 20221219-15:57:32]$ ./bootstrap-vcpkg.sh
Downloading vcpkg-glibc...
Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md
[mona@ARD-WS-24 20221219-15:57:38]$ ./vcpkg integrate install
Applied user-wide integration for this vcpkg root.
CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE=/home/mona/vcpkg/scripts/buildsystems/vcpkg.cmake"
[mona@ARD-WS-24 20221219-15:57:51]$ ./vcpkg install realsense2
Computing installation plan...
A suitable version of cmake was not found (required v3.25.0) Downloading portable cmake 3.25.0...
Downloading cmake...
https://github.com/Kitware/CMake/releases/download/v3.25.0/cmake-3.25.0-linux-x86_64.tar.gz->/home/mona/vcpkg/downloads/cmake-3.25.0-linux-x86_64.tar.gz
Extracting cmake...
The following packages will be built and installed:
  * libusb[core]:x64-linux -> 1.0.26#2
    realsense2[core]:x64-linux -> 2.51.1
  * vcpkg-cmake[core]:x64-linux -> 2022-10-30
  * vcpkg-cmake-config[core]:x64-linux -> 2022-02-06#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/mona/.cache/vcpkg/archives in 10.29 us. Use --debug to see more details.
Installing 1/4 libusb:x64-linux...
Building libusb[core]:x64-linux...
libusb currently requires the following tools and libraries from the system package manager:
    autoreconf
    libudev

These can be installed on Ubuntu systems via apt-get install autoreconf libudev-dev
-- Downloading https://github.com/libusb/libusb/archive/4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz -> libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz...
-- Extracting source /home/mona/vcpkg/downloads/libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz
-- Using source at /home/mona/vcpkg/buildtrees/libusb/src/3b7469543b-a464ae495e.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
CMake Error at scripts/cmake/vcpkg_configure_make.cmake:597 (message):
  libusb requires autoconf from the system package manager (example: "sudo
  apt-get install autoconf")
Call Stack (most recent call first):
  ports/libusb/portfile.cmake:61 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


error: building libusb:x64-linux failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libusb
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libusb]+Build+error
Include '[libusb] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: a22a71181 2022-12-19 (40 minutes ago)

![Screenshot from 2022-12-19 16-02-09](https://user-images.githubusercontent.com/1892917/208523359-0d181c90-d3b2-44cd-bb55-bd633910ade0.png)

Please use the prefilled template from /home/mona/vcpkg/installed/vcpkg/issue_body.md when reporting your issue.


```

It suggests to use `vcpkg update` and when I use it, I get this error:

```
[mona@ARD-WS-24 20221219-16:03:29]$ ./vcpkg update
Using local portfile versions. To update the local portfiles, use `git pull`.
No packages need updating.
```

When I do `git pull` nothing happens since I just got the repo.

```
[mona@ARD-WS-24 20221219-16:03:34]$ git pull
Already up to date.
```

```
[mona@ARD-WS-24 20221219-16:05:21]$ gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
[mona@ARD-WS-24 20221219-16:07:55]$ ./vcpkg version
vcpkg package management program version 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9

See LICENSE.txt for license information.
```

![Screenshot from 2022-12-19 16-02-09](https://user-images.githubusercontent.com/1892917/208524367-0a03a8ab-f7bf-4bec-89d0-df35ceaf6b48.png)

```
[mona@ARD-WS-24 20221219-16:09:27]$ sudo apt-get install autoreconf libudev-dev
[sudo] password for mona: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package autoreconf
```

However, I was able to install the following package:
```
[mona@ARD-WS-24 20221219-16:11:35]$ sudo apt -y install dh-autoreconf
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  autoconf automake autopoint autotools-dev libdebhelper-perl libltdl-dev libtool m4
Suggested packages:
  autoconf-archive gnu-standards autoconf-doc gettext libtool-doc gfortran | fortran95-compiler
  gcj-jdk m4-doc
The following NEW packages will be installed:
  autoconf automake autopoint autotools-dev dh-autoreconf libdebhelper-perl libltdl-dev libtool m4
0 upgraded, 9 newly installed, 0 to remove and 52 not upgraded.
Need to get 1,977 kB of archives.
After this operation, 7,325 kB of additional disk space will be used.
Get:1 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 m4 amd64 1.4.18-5ubuntu2 [199 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autoconf all 2.71-2 [338 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autotools-dev all 20220109.1 [44.9 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 automake all 1:1.16.5-1.3 [558 kB]
Get:5 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autopoint all 0.21-4ubuntu4 [422 kB]
Get:6 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libdebhelper-perl all 13.6ubuntu1 [67.2 kB]
Get:7 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libtool all 2.4.6-15build2 [164 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 dh-autoreconf all 20 [16.1 kB]
Get:9 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libltdl-dev amd64 2.4.6-15build2 [169 kB]
Fetched 1,977 kB in 0s (4,582 kB/s)      
Selecting previously unselected package m4.
(Reading database ... 280606 files and directories currently installed.)
Preparing to unpack .../0-m4_1.4.18-5ubuntu2_amd64.deb ...
Unpacking m4 (1.4.18-5ubuntu2) ...
Selecting previously unselected package autoconf.
Preparing to unpack .../1-autoconf_2.71-2_all.deb ...
Unpacking autoconf (2.71-2) ...
Selecting previously unselected package autotools-dev.
Preparing to unpack .../2-autotools-dev_20220109.1_all.deb ...
Unpacking autotools-dev (20220109.1) ...
Selecting previously unselected package automake.
Preparing to unpack .../3-automake_1%3a1.16.5-1.3_all.deb ...
Unpacking automake (1:1.16.5-1.3) ...
Selecting previously unselected package autopoint.
Preparing to unpack .../4-autopoint_0.21-4ubuntu4_all.deb ...
Unpacking autopoint (0.21-4ubuntu4) ...
Selecting previously unselected package libdebhelper-perl.
Preparing to unpack .../5-libdebhelper-perl_13.6ubuntu1_all.deb ...
Unpacking libdebhelper-perl (13.6ubuntu1) ...
Selecting previously unselected package libtool.
Preparing to unpack .../6-libtool_2.4.6-15build2_all.deb ...
Unpacking libtool (2.4.6-15build2) ...
Selecting previously unselected package dh-autoreconf.
Preparing to unpack .../7-dh-autoreconf_20_all.deb ...
Unpacking dh-autoreconf (20) ...
Selecting previously unselected package libltdl-dev:amd64.
Preparing to unpack .../8-libltdl-dev_2.4.6-15build2_amd64.deb ...
Unpacking libltdl-dev:amd64 (2.4.6-15build2) ...
Setting up libdebhelper-perl (13.6ubuntu1) ...
Setting up m4 (1.4.18-5ubuntu2) ...
Setting up autotools-dev (20220109.1) ...
Setting up autopoint (0.21-4ubuntu4) ...
Setting up autoconf (2.71-2) ...
Setting up automake (1:1.16.5-1.3) ...
update-alternatives: using /usr/bin/automake-1.16 to provide /usr/bin/automake (automake) in auto mo
de
Setting up libtool (2.4.6-15build2) ...
Setting up dh-autoreconf (20) ...
Setting up libltdl-dev:amd64 (2.4.6-15build2) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for install-info (6.8-4build1) ...

```

After installing the package below, the error above faded.

```
[mona@ARD-WS-24 20221219-16:11:35]$ sudo apt -y install dh-autoreconf
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  autoconf automake autopoint autotools-dev libdebhelper-perl libltdl-dev libtool m4
Suggested packages:
  autoconf-archive gnu-standards autoconf-doc gettext libtool-doc gfortran | fortran95-compiler
  gcj-jdk m4-doc
The following NEW packages will be installed:
  autoconf automake autopoint autotools-dev dh-autoreconf libdebhelper-perl libltdl-dev libtool m4
0 upgraded, 9 newly installed, 0 to remove and 52 not upgraded.
Need to get 1,977 kB of archives.
After this operation, 7,325 kB of additional disk space will be used.
Get:1 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 m4 amd64 1.4.18-5ubuntu2 [199 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autoconf all 2.71-2 [338 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autotools-dev all 20220109.1 [44.9 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 automake all 1:1.16.5-1.3 [558 kB]
Get:5 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 autopoint all 0.21-4ubuntu4 [422 kB]
Get:6 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libdebhelper-perl all 13.6ubuntu1 [67.2 kB]
Get:7 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libtool all 2.4.6-15build2 [164 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 dh-autoreconf all 20 [16.1 kB]
Get:9 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 libltdl-dev amd64 2.4.6-15build2 [169 kB]
Fetched 1,977 kB in 0s (4,582 kB/s)      
Selecting previously unselected package m4.
(Reading database ... 280606 files and directories currently installed.)
Preparing to unpack .../0-m4_1.4.18-5ubuntu2_amd64.deb ...
Unpacking m4 (1.4.18-5ubuntu2) ...
Selecting previously unselected package autoconf.
Preparing to unpack .../1-autoconf_2.71-2_all.deb ...
Unpacking autoconf (2.71-2) ...
Selecting previously unselected package autotools-dev.
Preparing to unpack .../2-autotools-dev_20220109.1_all.deb ...
Unpacking autotools-dev (20220109.1) ...
Selecting previously unselected package automake.
Preparing to unpack .../3-automake_1%3a1.16.5-1.3_all.deb ...
Unpacking automake (1:1.16.5-1.3) ...
Selecting previously unselected package autopoint.
Preparing to unpack .../4-autopoint_0.21-4ubuntu4_all.deb ...
Unpacking autopoint (0.21-4ubuntu4) ...
Selecting previously unselected package libdebhelper-perl.
Preparing to unpack .../5-libdebhelper-perl_13.6ubuntu1_all.deb ...
Unpacking libdebhelper-perl (13.6ubuntu1) ...
Selecting previously unselected package libtool.
Preparing to unpack .../6-libtool_2.4.6-15build2_all.deb ...
Unpacking libtool (2.4.6-15build2) ...
Selecting previously unselected package dh-autoreconf.
Preparing to unpack .../7-dh-autoreconf_20_all.deb ...
Unpacking dh-autoreconf (20) ...
Selecting previously unselected package libltdl-dev:amd64.
Preparing to unpack .../8-libltdl-dev_2.4.6-15build2_amd64.deb ...
Unpacking libltdl-dev:amd64 (2.4.6-15build2) ...
Setting up libdebhelper-perl (13.6ubuntu1) ...
Setting up m4 (1.4.18-5ubuntu2) ...
Setting up autotools-dev (20220109.1) ...
Setting up autopoint (0.21-4ubuntu4) ...
Setting up autoconf (2.71-2) ...
Setting up automake (1:1.16.5-1.3) ...
update-alternatives: using /usr/bin/automake-1.16 to provide /usr/bin/automake (automake) in auto mo
de
Setting up libtool (2.4.6-15build2) ...
Setting up dh-autoreconf (20) ...
Setting up libltdl-dev:amd64 (2.4.6-15build2) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for install-info (6.8-4build1) ...
```

However, now I have a new error:

```
[mona@ARD-WS-24 20221219-16:13:47]$ ./vcpkg install realsense2
Computing installation plan...
The following packages will be built and installed:
  * libusb[core]:x64-linux -> 1.0.26#2
    realsense2[core]:x64-linux -> 2.51.1
  * vcpkg-cmake[core]:x64-linux -> 2022-10-30
  * vcpkg-cmake-config[core]:x64-linux -> 2022-02-06#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/mona/.cache/vcpkg/archives in 7.622 us. Use --debug to see more details.
Installing 1/4 libusb:x64-linux...
Building libusb[core]:x64-linux...
libusb currently requires the following tools and libraries from the system package manager:
    autoreconf
    libudev

These can be installed on Ubuntu systems via apt-get install autoreconf libudev-dev
-- Using cached libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz.
-- Cleaning sources at /home/mona/vcpkg/buildtrees/libusb/src/3b7469543b-a464ae495e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/mona/vcpkg/downloads/libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz
-- Using source at /home/mona/vcpkg/buildtrees/libusb/src/3b7469543b-a464ae495e.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: /usr/bin/bash -c "V=1 ./../src/3b7469543b-a464ae495e.clean/configure  \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--prefix=/home/mona/vcpkg/installed/x64-linux/debug\" \"--bindir=\\${prefix}/../tools/libusb/debug/bin\" \"--sbindir=\\${prefix}/../tools/libusb/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/libusb\""
    Working Directory: /home/mona/vcpkg/buildtrees/libusb/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/mona/vcpkg/buildtrees/libusb/config-x64-linux-dbg-config.log
      /home/mona/vcpkg/buildtrees/libusb/config-x64-linux-dbg-out.log
      /home/mona/vcpkg/buildtrees/libusb/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:808 (vcpkg_execute_required_process)
  ports/libusb/portfile.cmake:61 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


error: building libusb:x64-linux failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libusb
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libusb]+Build+error
Include '[libusb] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: a22a71181 2022-12-19 (54 minutes ago)


Please use the prefilled template from /home/mona/vcpkg/installed/vcpkg/issue_body.md when reporting your issue.
```

Attached is the log for the newest installation error. Please let me know if you may need further information.
[config-x64-linux-dbg-config.log](https://github.com/microsoft/vcpkg/files/10262819/config-x64-linux-dbg-config.log)

**Host Environment**

- OS: [e.g. Windows/Linux etc...]
- Compiler: revision

**To Reproduce**

`./vcpkg install xxxx`

**Failure logs**

- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

https://github.com/IntelRealSense/librealsense/issues/11236

Here's the log file for latest error:
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/10262934/install-x64-linux-dbg-out.log)
and example of such error is:
```
/home/mona/vcpkg/buildtrees/realsense2/src/v2.51.1-1f2c91838a.clean/src/libusb/libusb.h:10:82: error: extended character ‘ is not valid in an identifier
```
I saw https://github.com/microsoft/vcpkg/issues/27977 but I have installed those two apt packages and still have error. Can you please guide me?

**Comment by monajalal** [comment]
error fixed by the following package:

```
[mona@ARD-WS-24 20221219-16:27:43]$ sudo apt-get install libudev-dev
```

and now I get a new error:

```
[mona@ARD-WS-24 20221219-16:26:29]$ ./vcpkg install realsense2
Computing installation plan...
The following packages will be built and installed:
  * libusb[core]:x64-linux -> 1.0.26#2
    realsense2[core]:x64-linux -> 2.51.1
  * vcpkg-cmake[core]:x64-linux -> 2022-10-30
  * vcpkg-cmake-config[core]:x64-linux -> 2022-02-06#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/mona/.cache/vcpkg/archives in 6.381 us. Use --debug to see more details.
Installing 1/4 libusb:x64-linux...
Building libusb[core]:x64-linux...
libusb currently requires the following tools and libraries from the system package manager:
    autoreconf
    libudev

These can be installed on Ubuntu systems via apt-get install autoreconf libudev-dev
-- Using cached libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz.
-- Cleaning sources at /home/mona/vcpkg/buildtrees/libusb/src/3b7469543b-a464ae495e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/mona/vcpkg/downloads/libusb-libusb-4239bc3a50014b8e6a5a2a59df1fff3b7469543b.tar.gz
-- Using source at /home/mona/vcpkg/buildtrees/libusb/src/3b7469543b-a464ae495e.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Installing x64-linux-dbg
-- Building x64-linux-rel
-- Installing x64-linux-rel
-- Fixing pkgconfig file: /home/mona/vcpkg/packages/libusb_x64-linux/lib/pkgconfig/libusb-1.0.pc
-- Fixing pkgconfig file: /home/mona/vcpkg/packages/libusb_x64-linux/debug/lib/pkgconfig/libusb-1.0.pc
-- Installing: /home/mona/vcpkg/packages/libusb_x64-linux/share/libusb/copyright
-- Performing post-build validation
Stored binary cache: "/home/mona/.cache/vcpkg/archives/e4/e457f9c979634d4ffad42b1ef8872ca4d288bcb9cde446330547efea2bdf4b78.zip"
Elapsed time to handle libusb:x64-linux: 10.67 s
Installing 2/4 vcpkg-cmake:x64-linux...
Building vcpkg-cmake[core]:x64-linux...
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg-port-config.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/copyright
-- Performing post-build validation
Stored binary cache: "/home/mona/.cache/vcpkg/archives/d2/d2c8ce9a8fe75ad354f98f6bcf1da44803208f07566710ee8a486275294bb5c5.zip"
Elapsed time to handle vcpkg-cmake:x64-linux: 10.1 ms
Installing 3/4 vcpkg-cmake-config:x64-linux...
Building vcpkg-cmake-config[core]:x64-linux...
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/vcpkg_cmake_config_fixup.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/vcpkg-port-config.cmake
-- Installing: /home/mona/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/copyright
-- Performing post-build validation
Stored binary cache: "/home/mona/.cache/vcpkg/archives/e4/e42cde407f3e66cb38574860ea72afa08edaa02e2ff0b8558bef83fdf94bf82d.zip"
Elapsed time to handle vcpkg-cmake-config:x64-linux: 12.25 ms
Installing 4/4 realsense2:x64-linux...
Building realsense2[core]:x64-linux...
-- Downloading https://github.com/IntelRealSense/librealsense/archive/v2.51.1.tar.gz -> IntelRealSense-librealsense-v2.51.1.tar.gz...
-- Extracting source /home/mona/vcpkg/downloads/IntelRealSense-librealsense-v2.51.1.tar.gz
-- Applying patch fix_openni2.patch
-- Applying patch fix_config_osx.patch
-- Using source at /home/mona/vcpkg/buildtrees/realsense2/src/v2.51.1-1f2c91838a.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: /home/mona/vcpkg/downloads/tools/cmake-3.25.0-linux/cmake-3.25.0-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /home/mona/vcpkg/buildtrees/realsense2/x64-linux-dbg
    See logs for more information:
      /home/mona/vcpkg/buildtrees/realsense2/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/realsense2/portfile.cmake:47 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building realsense2:x64-linux failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+realsense2
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[realsense2]+Build+error
Include '[realsense2] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: a22a71181 2022-12-19 (68 minutes ago)


Please use the prefilled template from /home/mona/vcpkg/installed/vcpkg/issue_body.md when reporting your issue.
```

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. This is a upstream issue https://github.com/IntelRealSense/librealsense/pull/10818, it has been fixed in latest version v2.53.1 of `realsense2`, please update vcpkg and rebuild `realsense2`.
